File: backend/spring-boot/src/main/java/org/bugzkit/api/user/repository/RoleRepository.java
package org.bugzkit.api.user.repository;

import java.util.List;
import java.util.Optional;
import java.util.Set;
import org.bugzkit.api.user.model.Role;
import org.bugzkit.api.user.model.Role.RoleName;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RoleRepository extends JpaRepository<Role, Long> {
  List<Role> findAllByNameIn(Set<RoleName> roleNames);

  Optional<Role> findByName(RoleName roleName);
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/user/repository/RoleRepository.kt
package org.bootstrapbugz.api.user.repository

import java.util.Optional
import org.bootstrapbugz.api.user.model.Role
import org.bootstrapbugz.api.user.model.Role.RoleName
import org.springframework.data.jpa.repository.JpaRepository

interface RoleRepository : JpaRepository<Role, Long> {
  fun findAllByNameIn(roleNames: Set<RoleName>): List<Role>

  fun findByName(roleName: RoleName): Optional<Role>
}
